src/kernels/conv3x3.s

```diff
@@ -729,32 +681,49 @@ miopenGcnAsmConv3x3U:
       s_addc_u32 s[weights_ptr+1], s[weights_ptr+1], 0
   .endif
 
+.macro s_mul_u64_u32 dst_lo, dst_hi, src0, src1, vtmp0, vtmp1
+    v_mov_b32 \vtmp0, \src1
+    v_mul_hi_u32 \vtmp1, \src0, \vtmp0
+    v_mul_lo_u32 \vtmp0, \src0, \vtmp0
+    v_readfirstlane_b32 \dst_hi, \vtmp1
+    v_readfirstlane_b32 \dst_lo, \vtmp0
+.endm
   // construct input buffer descriptor
+vtmp = linesA
+stmp2 = in_desc+2
+  s_mov_b32 s[tmp], 0
   .if batch_size > 1
-    s_mul_i32 s[tmp], s[gid_z], input_feature_map_stride * input_channels
-    s_add_u32 s[in_desc], s[in_desc], s[tmp] // add input image batch offset
-    s_addc_u32 s[in_desc+1], s[in_desc+1], 0
+    // adds (s[gid_z] * input_feature_map_stride * input_channels)
+    // to input base addr
+    s_mul_i32 s[tmp], s[gid_z], input_channels
   .endif
-  s_mov_b32 s[in_desc+2], input_buffer_window // size
-  s_mov_b32 s[in_desc+3], 0x00027000
-
 .if group_counts > 1
-  s_mul_i32 s[tmp], s[s_group_id], c_group_size * input_feature_map_stride // c_group_offset
-  s_add_u32 s[in_desc], s[in_desc], s[tmp]
-  s_addc_u32 s[in_desc+1], s[in_desc+1], 0
+    // adds (s[s_group_id] * input_feature_map_stride * c_group_size)
+    // to input base addr
+    s_mul_i32  s[stmp2], s[s_group_id], c_group_size
+    s_add_u32 s[tmp], s[tmp], s[stmp2]
+.endif
+.if batch_size > 1 || group_counts > 1
```

Comment:
Configs with `group_count != 1` must be tested.

Reply:
Thank you! You can find these test cases in the updated PR description.